datasources.go

```diff
@@ -1,25 +1,60 @@
+	Automatic *InvoiceHandlingMode `json:"automatic,omitempty"`
+}
+
+// InvoiceHandlingMode represents the configuration for invoice handling mode.
+type InvoiceHandlingMode struct {
+	CreateSubscriptionWhenInvoiceIs         string `json:"create_subscription_when_invoice_is"`
+	UpdateSubscriptionWhenInvoiceIs         string `json:"update_subscription_when_invoice_is"`
+	PreventSubscriptionForInvoiceVoided     bool   `json:"prevent_subscription_for_invoice_voided"`
+	PreventSubscriptionForInvoiceRefunded   bool   `json:"prevent_subscription_for_invoice_refunded"`
+	PreventSubscriptionForInvoiceWrittenOff bool   `json:"prevent_subscription_for_invoice_written_off"`
+}
+
 // DataSource represents API data source in ChartMogul.
 // See https://dev.chartmogul.com/v1.0/reference#list-data-sources
 type DataSource struct {
-	UUID      string `json:"uuid"`
-	Name      string `json:"name"`
-	CreatedAt string `json:"created_at"`
-	Status    string `json:"status"`
-	System    string `json:"system"`
-	Errors    Errors `json:"errors,omitempty"`
+	UUID                         string                  `json:"uuid"`
+	Name                         string                  `json:"name"`
+	CreatedAt                    string                  `json:"created_at"`
+	Status                       string                  `json:"status"`
+	System                       string                  `json:"system"`
+	ProcessingStatus             *ProcessingStatus       `json:"processing_status,omitempty"`
+	AutoChurnSubscriptionSetting *bool                   `json:"auto_churn_subscription_setting,omitempty"`
+	InvoiceHandlingSetting       *InvoiceHandlingSetting `json:"invoice_handling_setting,omitempty"`
+	Errors                       Errors                  `json:"errors,omitempty"`
 }
 
 // DataSources is the result of listing data sources, but doesn't contain any paging.
 type DataSources struct {
 	DataSources []*DataSource `json:"data_sources"`
 }
 
+// ExtraDataSourceParams are optional parameters for additional data source information.
+type ExtraDataSourceParams struct {
+	WithProcessingStatus             *bool `json:"with_processing_status,omitempty"`
+	WithAutoChurnSubscriptionSetting *bool `json:"with_auto_churn_subscription_setting,omitempty"`
+	WithInvoiceHandlingSetting       *bool `json:"with_invoice_handling_setting,omitempty"`
+}
+
 // ListDataSourcesParams are optional parameters for listing data sources.
 type ListDataSourcesParams struct {
-	Name   string `json:"name,omitempty"`
-	System string `json:"system,omitempty"`
+	Name                             string `json:"name,omitempty"`
+	System                           string `json:"system,omitempty"`
+	WithProcessingStatus             *bool  `json:"with_processing_status,omitempty"`
+	WithAutoChurnSubscriptionSetting *bool  `json:"with_auto_churn_subscription_setting,omitempty"`
+	WithInvoiceHandlingSetting       *bool  `json:"with_invoice_handling_setting,omitempty"`
```

Comment:
I'd like the `ListDataSourcesParams` to somehow extend the `ExtraDataSourceParams`, to keep the code DRY, but I couldn't figure out how.

If you do know how, please let me know - thanks.

Reply:
Does embedded struct not work for some reason?
https://www.geeksforgeeks.org/go-language/embedded-structures-in-golang/